### PR TITLE
Explain what the range enum defined here is for

### PR DESCRIPTION
**Pull request by jamesnvc**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,3 +2,43 @@
 
 This plugin contains the ranges used by a number of Kaleidoscope plugins. Mostly
 useful for developers, but a dependency of other plugins, too.
+
+If you're a plugin developer and need to know how you can make keycodes for just your plugin to consume, or are otherwise curious what this is for, read one.
```
Comment (algernon) [inline README.md]: Should that `read one` at the end be `read on` instead?

File: README.md [hunk 2]
```diff
@@ -2,3 +2,43 @@
...
+  uint16_t raw;
+// bunch of operator overloads elided...
+} Key;
+```
+That is, a key is a 16-bit integer that, by default, has 8 bits for the keyCode and 8 bits for flags.
+For normal keypresses, this is straightforward: the 8-bit keyCode is the normal HID code corresponding to a normal keyboard key and the flags are used to indicate the modifiers being held.
+
+However, many plugins want to be able to make key presses do something special and need some way of indicating that a key is not just an ordinary key.
+To do this, we take advantage of the fact that our flags field is 8 bits, but there are only five normal modifiers (control, shift, GUI, right alt, and left alt).
+<!-- question: does bit six have any signicance? it seems to be unused
```
Comment (algernon) [inline README.md]: When the `RESERVED` bit is set, the other 15 bits have no special meaning anymore. When that is set, we only use the `key.raw` part of the union, the `flags` and `keyCode` are unused in this case.

Does that clear things up?

File: README.md [hunk 3]
```diff
@@ -2,3 +2,43 @@
...
+--> Therefore, we can use a bit to indicate that something special is happening with a given key:
+In this case, by setting the high bit of the flags field, we indicate that this is a reserved key and isn't going to be interpreted normally.
+
+This way, a plugin can make a key be sent with the high bit of the flags field set, then it is free to use the rest of the 16 bits as it sees fit and be assured that it won't conflict with the built-in keys.
+
+However, there is a problem with this:
+Many plugins will want to do this, so how can they be sure that the key codes they make won't be interpreted by another plugin?
+Thus, we come to the purpose of this enum: 
+The `range` enum gives the ranges of possible raw key values that each plugin will use; by referring to it, plugins can be sure none of them will step on each others' toes.
+
```
Comment (algernon) [inline README.md]: My issue with this explanation is that... it is similar to how I'd explain it, and... well. It doesn't make much sense, isn't all that easy to follow.

Unfortunately, I have no better way to propose, either. (Will try to explain my issues later, a bit short on time at the moment)

**Comment by obra** [comment]
Apologies for over-zealous merging.

File: README.md (hunk 1 above)
Comment (jamesnvc) [reply README.md]: oops, yes indeed

File: README.md (hunk 2 above)
Comment (jamesnvc) [reply README.md]: Oh, I was just noticing that bits 1-5 are for modifiers, bit 7 is injected, bit 8 is reserved but bit 6 has no meaning.

File: README.md (hunk 3 above)
Comment (jamesnvc) [reply README.md]: Maybe we could just provide a few examples of normal key press, TapDance key, OneShot key?

Comment (jamesnvc) [reply README.md]: Presumably the only people reading this are plugin authors that have *some* context to all this stuff?

**Comment by algernon** [comment]
No worries. Some docs are better than none, and docs are the easiest to improve!

**Pull request merged**